job/replay_manager.go

```diff
@@ -37,7 +40,9 @@ type ReplayManagerConfig struct {
 
 type ReplayManager interface {
 	Init()
-	Replay(context.Context, *models.ReplayWorkerRequest) (string, error)
+	Replay(context.Context, *models.ReplayRequest) (string, error)
```

Comment:
Any reason we are using pointer in the request?

Reply:
was read somewhere that instead of passing the whole struct, better practice is to reference it. let me know your thoughts & consideration?

Reply:
If the struct is large then only it makes sense, in these cases, to pass it via reference, else it is actually bad for cache coherence. CPU works with L1, L2, L3... caches, each of them has different sizes, it's actually faster to copy a small struct and keep it in the cache then to dereference a pointer.

Reply:
Thank you for the explanation. I have updated to not use pointer in replay requests.